docs/source/features/plugins.md

```diff
@@ -0,0 +1,320 @@
+
+    /* A plugin which is defined in-line. */
+    {
+      /* ... plugin event hooks ... */
+    },
+  ],
+})
+```
+
+## Definition
+
+> **Types:** To facilitate plugin development, the `apollo-server-plugin-base` module exports [the `ApolloServerPlugin` interface](https://github.com/apollographql/apollo-server/blob/master/packages/apollo-server-plugin-base/src/index.ts) for plugins to utilize.  It's recommended to use this interafce when building custom plugins.
+
+
+A plugin defines the life-cycle events it wishes to act upon using an object which maps events (specific events are defined in further detail later in this document) to the functions that implement them.
+
+For example, here is a very simple plugin, defined as an object literal, which implements a `requestDidStart` event:
+
+```js
+{
+  requestDidStart() {
+    console.log('The request started.');
+  },
+}
+```
+
+This plugin might be directly included as an element of the `plugins`, or it could be provided as a separate module:
+
+```js
+module.exports = {
+  requestDidStart() {
+    /* ... */
+  },
+};
+```
+
+Plugins which accepted options might providing a function which returns an object that implements a object matching the `ApolloServerPlugin` interface:
+
+```js
+/* localPluginModule.js */
+module.exports = (options) => {
+  /* ...Plugin specific implementation... */
+
+  return {
+    requestDidStart() {
+      console.log('The options were', options);
+    },
+  };
+};
+```
+
+Within the `plugins` array, this `localPluginModule.js` would be used as:
+
+```js
+  /* ... Existing, required ApolloServer configuration. ... */
+
+  plugins: [
+
+    require('./localPluginModule')({
+      /* ...configuration options, when necessary! */
+    }),
+
+  ],
+
+  /* ... any additional ApolloServer configuration. ... */
+```
+
+And finally, advanced cases can implement the `ApolloServerPlugin` interface via a factory function.  The factory function will receive `pluginInfo`, which can allow implementors to adjust their behavior based on circumstantial factors:
+
+```js
+/* advancedPluginModule.js */
+module.exports = (options) => {
+  /* ...Plugin specific implementation... */
+
+  return (pluginInfo) => {
+    console.log('The pluginInfo was', pluginInfo);
+
+    return {
+      requestDidStart() {
+        console.log('The options were', options);
+      }
+    };
+  };
+}
+```
+
+And again, this could be used as a plugin by defining it in the `plugins` array:
+
+```js
+  /* ... Existing, required ApolloServer configuration. ... */
+
+  plugins: [
+
+    require('./advancedPluginModule')({
+      /* ...configuration options, when necessary! */
+    }),
+
+  ],
+
+  /* ... any additional ApolloServer configuration. ... */
+```
+
+> **Note:** Currently the `pluginInfo` is undefined, but future additions to the plugin API will enable this functionality.  For now, the factory function facilities are in place to use, but `pluginInfo` is simply not available.
```

Comment:
Let's comment out the pluginInfo section with a todo until it is available. When do you anticipate we'll be able to use that functionality?

Reply:
Not sure!  It's reserved for future use, though it probably makes sense to get something — anything — in there sooner rather than later! 😄

Reply:
Removed for now!